src/models/SearchResultsBib.ts

```diff
@@ -0,0 +1,81 @@
+import type { SearchResult } from "../types/searchTypes"
+import type { ElectronicResource } from "../types/bibTypes"
+
+export default class SearchResultsBib {
+  id: string
+  title: string
+  yearPublished?: string
+  materialType?: string
+  publicationStatement?: string
+  eResources?: ElectronicResource[]
+  numPhysicalItems: number
+
+  constructor(result: SearchResult) {
+    this.id = result["@id"] ? result["@id"].substring(4) : ""
+    this.title = this.getTitleFromResult(result)
+    this.yearPublished = this.getYearFromResult(result)
+    this.materialType = result.materialType[0]?.prefLabel || null
+    this.publicationStatement = result.publicationStatement?.length
+      ? result.publicationStatement[0]
+      : null
+    this.eResources = result.electronicResources || null
+    this.numPhysicalItems = result.numItemsTotal || 0
+  }
+
+  get url() {
+    return `/bib/${this.id}`
+  }
+
+  get numElectronicResources() {
+    return this.eResources?.length || 0
+  }
+
+  get hasPhysicalItems() {
+    return this.numPhysicalItems > 0
+  }
+
+  get numItems() {
```

Comment:
would love to discuss this definition. we've ended up in some snarls in the past due to confusion about what constitutes the number of items. In this definition, num electronic resources will not be a part of the count of items unless there are NO physical items. That is helpful for determining whether to display the items table, but not helpful for displaying item counts.
If this numItems property exists to determine whether we render the items table or not, then maybe we could create a separate property (thats a big question mark). 
Lets circle back on this one when I'm back from vacation. There is potentially some product questions here as well regarding full digital representations of works vs supplemental content like table of contents. 
If we really iron this out and document i hope we can avoid bringing legacy headaches into this new repo!